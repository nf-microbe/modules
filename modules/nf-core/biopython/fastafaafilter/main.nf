process BIOPYTHON_FASTAFAAFILTER {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta) , path(fasta)
    tuple val(meta2), path(faa)

    output:
    tuple val(meta), path("${prefix}.filtered.fasta.gz")    , emit: fasta
    tuple val(meta), path("${prefix}.filtered.faa.gz")      , emit: faa
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    fastafaafilter.py \\
        --input_fasta ${fasta} \\
        --input_faa ${faa} \\
        --prefix ${prefix}.filtered \\
        ${args}

    gzip ${prefix}.filtered.fasta ${prefix}.filtered.faa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo "" | gzip > ${prefix}.filtered.fasta.gz
    echo "" | gzip > ${prefix}.filtered.faa.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
    END_VERSIONS
    """
}
