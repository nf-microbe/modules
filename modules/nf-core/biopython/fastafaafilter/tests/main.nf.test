nextflow_process {

    name "Test Process: BIOPYTHON_FASTAFAAFILTER"
    script "../main.nf"
    process "BIOPYTHON_FASTAFAAFILTER"
    config "./nextflow.config"

    tag "modules_nfcore"
    tag "modules"
    tag "biopython"
    tag "biopython/fastafaafilter"

    test("fasta.gz + faa.gz") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/illumina/faa/test1.pyrodigalgv.faa.gz", checkIfExists: true),
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkIfExists: true),
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("stub") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/prokaryotes/bacteroides_fragilis/illumina/faa/test1.pyrodigalgv.faa.gz", checkIfExists: true),
                ]
                input[1] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + "genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz", checkIfExists: true),
                ]
                """
            }
        }
        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
