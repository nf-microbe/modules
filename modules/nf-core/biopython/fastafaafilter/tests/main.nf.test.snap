{
    "fasta.gz + faa.gz": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.filtered.fasta.gz:md5,238c9c4b8fd2a3f067bf3e840ead459a"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.filtered.faa.gz:md5,fbcb4274ae897b730beff303fbf2c20c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,4e4d46ab9a84de4fc18e4c76ba321b25"
                ],
                "faa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.filtered.faa.gz:md5,fbcb4274ae897b730beff303fbf2c20c"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.filtered.fasta.gz:md5,238c9c4b8fd2a3f067bf3e840ead459a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4e4d46ab9a84de4fc18e4c76ba321b25"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-27T22:40:02.177316594"
    },
    "stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.filtered.fasta.gz:md5,238c9c4b8fd2a3f067bf3e840ead459a"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.filtered.faa.gz:md5,fbcb4274ae897b730beff303fbf2c20c"
                    ]
                ],
                "2": [
                    "versions.yml:md5,4e4d46ab9a84de4fc18e4c76ba321b25"
                ],
                "faa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.filtered.faa.gz:md5,fbcb4274ae897b730beff303fbf2c20c"
                    ]
                ],
                "fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.filtered.fasta.gz:md5,238c9c4b8fd2a3f067bf3e840ead459a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4e4d46ab9a84de4fc18e4c76ba321b25"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.1",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-27T22:40:12.534557805"
    }
}